plugins/GSdx/GSSettingsDlg.cpp

```diff
@@ -276,6 +276,10 @@ bool GSSettingsDlg::OnCommand(HWND hWnd, UINT id, UINT code)
 			if(ComboBoxGetSelData(IDC_RENDERER, data))
 			{
 				theApp.SetConfig("Renderer", (int)data);
+				if ((data == GSRendererType::OGL_HW || data == GSRendererType::OGL_SW || data == GSRendererType::OGL_OpenCL) && (theApp.GetConfigI("UserHacks_HalfPixelOffset")>1))
+				{
+					theApp.SetConfig("UserHacks_HalfPixelOffset", 1);
```

Comment:
I think you only need to check for ``OGL_HW`` and by the way, I don't think you need this in the first place. Even if you have a value above 1 for ``UserHacks_HalfPixelOffset`` in INI file, nothing unexpected will happen when running the D3D renderer, the normal HPO hack will take place as ``UserHacks_HalfPixelOffset`` is a not a zero value. (Refer the code in TextureCache for more information)